### PR TITLE
[move-prover][doc] update spec lang doc on `old(..)` in inline specs

### PR DESCRIPTION
**Pull request by mengxu-fb**
The support for `old(..)` expressions in `assert` and `assume`
statements is landed. So update the doc for consistency.

[Preview of the new doc](https://github.com/diem/diem/blob/f2327e3e6d4d5d651dfa2c235fca872a618a3f0f/language/move-prover/doc/user/spec-lang.md)

## Motivation

update doc for consistency

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

eyes

### CODE REVIEW

File: language/move-prover/doc/user/spec-lang.md [hunk 1]
```diff
@@ -1032,6 +1039,70 @@ A loop invariant may comprise both assert and assume statements. The assume stat
...
+}
+```
+
+Note the usage of `old(v)` in the loop invariants. Without them, it is hard to express the
+invariant that the vector is partially reversed while the loop is iterating and the rest
+remain unchanged.
+
+However, unlike the `old(T)` expressions in `ensures` conditions where `T` can be any valid
+expression (e.g., `old(v[i])` is allowed), the `old(T)` expressions in `assert` and `assumes`
+statements only accept a single expression as `T` and that expression must be a function argument of
```
Comment (shazqadeer) [inline language/move-prover/doc/user/spec-lang.md:1102]: By "single expression", do you mean "single variable"?

**Review by shazqadeer — APPROVED** [review]

File: language/move-prover/doc/user/spec-lang.md (hunk 1 above)
Comment (mengxu-fb) [reply language/move-prover/doc/user/spec-lang.md:1102]: Yes, a "single variable". I'll make an update.

Comment (mengxu-fb) [reply language/move-prover/doc/user/spec-lang.md]: updated in the fixup! commit.

**Review by shazqadeer — DISMISSED** [review]

**Pull request closed without merging**